### ISSUE TITLE
BidirectionSearchTest extended to include environments with different actions from adjoining nodes

### ISSUE THREAD

**Issue by ctjoreilly** [aimacode/aima-java#312]
This is related to the now closed  #298. Basically, the BidirectionalSearchTest needs to be extended with tests using an environment that ensure the sequence of actions from start to goal make sense, i.e. in the example if you are in a grid and the actions are left, right, up, down and say you start at the top and bottom and meet in the middle the actions would be something like:

start->meet at = down, down
goal->meet at = up up

Simply joining the lists of actions will not work in this scenario (unlike the current Map2D tests). The implementations of BidirectionalSearch need to be tested to ensure they handle this correctly.

**Comment by globalworming** [comment]
A basic grid-like environment should be easy. I'd say this environment needs walls like http://www.redblobgames.com/pathfinding/tower-defense/
I'll look into that when the monte carlo localization frustrates me again :)

**Comment by HasinduGajanayake** [comment]
I am working on this enhancement

**Comment by ctjdev** [comment]
@HasinduGajanayake  can you explain me about the enhancement?

**Comment by HasinduGajanayake** [comment]
Yeah, Its a pleasure to tell you about this enhancement. there is a test class called BidirectionalSearchTest.But its not optimized yet.Think  if you are in a grid and the actions are left, right, up, down and say you start at the top and bottom and meet in the middle the actions would be 
start->meet at = down, down
goal->meet at = up up
But the thing is simply joining the actions will not work..
